pkg/proc/interface.go

```diff
@@ -134,3 +139,18 @@ func (cctx *ContinueOnceContext) GetManualStopRequested() bool {
 	defer cctx.StopMu.Unlock()
 	return cctx.manualStopRequested
 }
+
+// NumProcs returns the number of processes in the target group being resumed.
+func (cctx *ContinueOnceContext) NumProcs() int {
+	return len(cctx.grp.targets)
+}
+
+// Proc returns the i-th process in the group.
+func (cctx *ContinueOnceContext) Proc(i int) ProcessInternal {
```

Comment:
This also feels like it should be on `TargetGroup`.

Reply:
> This might actually be a good use case for generics

I don't want to raise the minimum version of Go this much just for this. 

Also I'm not sure how to do it in a way that doesn't suck more than this. If we add a type parameter to ContinueOnceContext then Target and TargetGroup also need to have it. This already sucks because Target and TargetGroup don't actually care about the specific type of the processes they contain besides that they satisfy an interface. On top of that TargetGroup sits inside Debugger and the specific type of the process it contains is determined at runtime, so we woud have to hide the type parameter through an interface to store the TargetGroup inside Debugger.

A better way to avoid the typecast is to keep two parallel slices, one of Targets and one of Processes. TargetGroup would contain a ProcessGroup interface:

```
type ProcessGroup interface {
    NumTargets() int
    Target(i int) *Target
}
```

which would be implemented like this:

```
type nativeProcessGroup struct {
    targets []*proc.Target
    processes []*nativeProcess
}
```

and the targets and processes have to be kept in sync at all times.

Reply:
You're right, introducing generics would be a pretty big rewrite of the types and plumbing it through would be cumbersome. Even if we did want to do that eventually, it wouldn't make sense to add that much noise to the PR. Let's leave the type casting for now until we can figure out a more elegant solution (I don't like the double-list approach).